llvm/lib/Transforms/IPO/SampleProfileMatcher.cpp

```diff
@@ -328,8 +405,8 @@ void SampleProfileMatcher::recordCallsiteMatchStates(
   // IR callsites.
   for (const auto &I : ProfileAnchors) {
     const auto &Loc = I.first;
-    [[maybe_unused]] const auto &Callees = I.second;
-    assert(!Callees.empty() && "Callees should not be empty");
+    [[maybe_unused]] StringRef CalleeName = I.second.stringRef();
+    assert(!CalleeName.empty() && "Callees should not be empty");
```

Comment:
nit: `CalleeName` is only used in assertion, why not fold it into the assertion?